### PR TITLE
PLFM-6678.2

### PR DESCRIPTION
**Pull request by brucehoff**
This PR introduces a new "login" endpoint that returns an access token (as opposed to a session token).

### CODE REVIEW

File: services/repository/src/main/java/org/sagebionetworks/auth/controller/AuthenticationController.java [hunk 1]
```diff
@@ -64,9 +64,9 @@
  * initially, obtaining a session token for use in other requests.
  * </p>
  * <p>
- * To authenticate using the session token returned by the 
- * <a href="${POST.login}">POST /login</a> service,
- * add it to the request a header named "sessionToken".
+ * To authenticate using the access token returned by the 
+ * <a href="${POST.login2}">POST /login</a> service,
```
Comment (marcomarasca) [inline services/repository/src/main/java/org/sagebionetworks/auth/controller/AuthenticationController.java]: Might want to update the /login endpoint

File: services/repository/src/main/java/org/sagebionetworks/auth/controller/AuthenticationController.java [hunk 2]
```diff
@@ -123,21 +123,32 @@ Session authenticate(
...
-	 * invalidated. The user must accept the terms of use before a session token
-	 * is issued.
+	 * Retrieve an access token that will be usable for 24 hours. 
+	 * The user must accept the terms of use before the access token
+	 * can be used.
 	 */
 	@RequiredScope({})
 	@ResponseStatus(HttpStatus.CREATED)
-	@RequestMapping(value = UrlHelpers.AUTH_LOGIN, method = RequestMethod.POST)
+	@RequestMapping(value = UrlHelpers.AUTH_LOGIN_2, method = RequestMethod.POST)
```
Comment (marcomarasca) [inline services/repository/src/main/java/org/sagebionetworks/auth/controller/AuthenticationController.java:146]: Can we double check if a "standard" endpoint name convention exists? E.g. what other services use or maybe specs?

File: lib/models/src/main/java/org/sagebionetworks/repo/model/AuthorizationConstants.java [hunk 3]
```diff
@@ -200,6 +200,11 @@ public static boolean isBootstrapPrincipalId(long id) {
 	 */
 	public static final String SYNAPSE_OAUTH_CLIENT_ID = "0";
 	
+	/**
+	 * The time, in seconds, that an access token lasts
+	 */
+	public static final long ACCESS_TOKEN_EXPIRATION_TIME_SECONDS = 3600*24L; // a day
```
Comment (marcomarasca) [inline lib/models/src/main/java/org/sagebionetworks/repo/model/AuthorizationConstants.java:206]: Please double check with Aaron about the recommended expiration timeout for this kind of access token (if any).

File: services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/authentication/AuthenticationManagerImpl.java [hunk 4]
```diff
@@ -225,7 +229,27 @@ public LoginResponse login(LoginRequest request){
 
 		validateAuthReceiptAndCheckPassword(userId, password, authenticationReceipt);
 
-		return getLoginResponseAfterSuccessfulPasswordAuthentication(userId);
+		return getLoginResponseWithSessionAfterSuccessfulPasswordAuthentication(userId);
+	}
+
+	/**
+	 * Note: We explicitly removed the transaction annotation from this method (see:
```
Comment (marcomarasca) [inline services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/authentication/AuthenticationManagerImpl.java]: Maybe remove the comment :)

**Review by marcomarasca — APPROVED** [review]

File: lib/models/src/main/java/org/sagebionetworks/repo/model/AuthorizationConstants.java (hunk 3 above)
Comment (brucehoff) [reply lib/models/src/main/java/org/sagebionetworks/repo/model/AuthorizationConstants.java:206]: In progress (via Jira).

File: services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/authentication/AuthenticationManagerImpl.java (hunk 4 above)
Comment (brucehoff) [reply services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/authentication/AuthenticationManagerImpl.java]: Done!

File: services/repository/src/main/java/org/sagebionetworks/auth/controller/AuthenticationController.java (hunk 2 above)
Comment (brucehoff) [reply services/repository/src/main/java/org/sagebionetworks/auth/controller/AuthenticationController.java:146]: The [OpenID Connect discovery document spec'](https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderMetadata) has no definition of the authentication endpoint.  I think we can use what we wish.  I am following the convention of our [`wiki2`](https://rest-docs.synapse.org/rest/#org.sagebionetworks.repo.web.controller.V2WikiController) and [`entitybundle2`](https://rest-docs.synapse.org/rest/#org.sagebionetworks.repo.web.controller.EntityBundleV2Controller) services, i.e. putting a '2' at the end of the URI.

**Comment by brucehoff** [comment]
@marcomarasca   Thanks for the CR/approval.  I made the two small changes you asked for.  My question about the access token expiration is pending.  I will merge this PR in the meantime and incorporate any change to the expiration in a later PR.

**Pull request merged**